client/blocks/upe/fields.js

```diff
@@ -63,6 +89,122 @@ const UPEField = ( {
 		createIntent();
 	}, [ paymentIntentId, hasRequestedIntent, api, errorMessage ] );
 
+	const customerData = useCustomerData();
+
+	useEffect( () => {
+		if (
+			paymentMethodsConfig.link !== undefined &&
+			paymentMethodsConfig.card !== undefined
+		) {
+			const shippingAddressFields = {
+				line1: 'shipping-address_1',
+				line2: 'shipping-address_2',
+				city: 'shipping-city',
+				state: 'components-form-token-input-1',
+				postal_code: 'shipping-postcode',
+				country: 'components-form-token-input-0',
+			};
+			const billingAddressFields = {
+				line1: 'billing-address_1',
+				line2: 'billing-address_2',
+				city: 'billing-city',
+				state: 'components-form-token-input-3',
+				postal_code: 'billing-postcode',
+				country: 'components-form-token-input-2',
+			};
+
+			const appearance = getAppearance();
+			elements.update( { appearance } );
+
+			enableStripeLinkPaymentMethod( {
+				api,
+				elements,
+				emailId: 'email',
+				fill_field_method: ( address, nodeId, key ) => {
+					const setAddress =
+						shippingAddressFields[ key ] === nodeId
+							? customerData.setShippingAddress
+							: customerData.setBillingData;
+					const customerAddress =
+						shippingAddressFields[ key ] === nodeId
+							? customerData.shippingAddress
+							: customerData.billingData;
+
+					if ( undefined === customerAddress ) {
+						return;
```

Comment:
@mordeth `customerAddress` is `undefined` for me. For this reason,  the address fields are not populated by StripeLink:

<img width="1137" alt="Screen Shot 2022-07-19 at 10 19 19" src="https://user-images.githubusercontent.com/1553182/179703298-ed987174-c595-4169-a02c-cc62da7d2fa4.png">

Reply:
@dpaun1985 @mordeth I am still able to reproduce this issue after latest changes. 

I've been digging into the code and I manage to make it work with a couple of changes. I'll add some comments to the lines I changed